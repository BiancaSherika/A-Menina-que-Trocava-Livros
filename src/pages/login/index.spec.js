import { Login } from './index.js';
import * as services from '../../services/index.js';
import * as utils from '../../utils/history.js';

services.signIn = jest.fn(() => Promise.resolve(true));
services.loginWithGoogle = jest.fn(() => Promise.resolve(true));
services.registerUser = jest.fn(() => Promise.resolve(true));
utils.onNavigate = jest.fn();

describe('Login', () => {
    it('should be a function', () => {
        expect(typeof Login).toBe('function');
    }); 

    it('should load the page', () => {
        expect(Login()).toMatchSnapshot();
    });

    it('When the user clicks the button, he should call sign with email and password', () => {
        Login().querySelector('#button-login').dispatchEvent(new Event('click'));
        expect(services.signIn).toHaveBeenCalled();
    });

    it('When the user clicks the button he should call sign with google account', () => {
        Login().querySelector('#button-google').dispatchEvent(new Event('click'));
        expect(services.loginWithGoogle).toHaveBeenCalled();
    });

    it('When the user clicks the button, they should redirect to the registration page', () => {
        Login().querySelector('#button-create-account').dispatchEvent(new Event('click'));
        expect(utils.onNavigate).toHaveBeenCalledWith('/register');
    });
});