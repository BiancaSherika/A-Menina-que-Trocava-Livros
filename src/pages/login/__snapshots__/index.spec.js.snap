// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should load the page 1`] = `
<div>
  
    
  <div
    <div=""
    class="container"
  >
    
        
    <form
      class="container-login"
    >
      
          
      <h1
        class="title"
      >
        A Menina que Trocava Livros
      </h1>
                    
          
      <input
        class="input-login"
        id="e-mail"
        placeholder="E-mail"
        type="email"
      />
      
          
      <input
        class="input-login"
        id="my-password"
        placeholder="Senha"
        type="password"
      />
             
          
      <button
        class="button-login"
        id="button-login"
        type="submit"
      >
        Entrar
      </button>
      
          
      <button
        class="button-create-account"
        id="button-create-account"
        type="submit"
      >
        Criar Conta
      </button>
      
          
      <button
        class="button-google"
        id="button-google"
      >
        <img
          alt="img-google"
          id="img-google"
          src="img/google.png"
        />
      </button>
      
          
      <label
        class="google"
      >
        Entrar com o Google
      </label>
      
          
      <script
        src="./register/index.js"
        type="module"
      />
      
        
    </form>
    

        
    <img
      <a=""
      alt="Logo"
      class="container-logo"
      href="https://www.flaticon.com/br/autor"
      src="img/livro.png"
    />
    

      
  </div>
  
    

    
  <footer>
    
    
    <ul
      class="footer"
    >
      
      
      <a
        href="https://github.com/BiancaSherika"
        target="_blank"
      >
        <strong>
          @Bianca Sherika
        </strong>
      </a>
      
      
      <a
        href="https://github.com/ly-cardozo"
        target="_blank"
      >
        <strong>
          @Eliane Cardozo
        </strong>
      </a>
      
      
      <a
        href="https://github.com/gabrielasilva1991"
        target="_blank"
      >
        <strong>
          @Gabriela Silva
        </strong>
      </a>
      
      
      <a
        href="https://www.laboratoria.la/"
        target="_blank"
      >
        <strong>
          by Laborat√≥ria
        </strong>
      </a>
      
    
    </ul>
    
    
  </footer>
  
    
</div>
`;
